src/main/java/LineTokenizer.java

```diff
@@ -0,0 +1,29 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class LineTokenizer {
+    Map<String, ArrayList<Integer>> newMapWords;
+    List<Integer> linesWithWord;
+
+    public Map<String, ArrayList<Integer>> mapWords(List<String> lines) {
+        newMapWords = new HashMap<>();
+        lines.stream()
+                .map(line -> line.trim().split("\\s+"))
+                .flatMap(Stream::of)
+                .map(String::toUpperCase).filter(word -> !newMapWords.containsKey(word))
+                .forEach(word -> newMapWords.put(word, (ArrayList<Integer>) getLinesWithWord(lines, word)));
```

Comment:
Can you use the `.collect` instead of `.forEach()`?

Reply:
@Ksef you can do something like this 
```
.flatMap(Stream::of)
.collect(Collectors.toMap(
                        Function.identity(),
                        word -> getLinesWithWord(lines, word),
                        (existing, replacement) -> existing));
```
then remove this map and return the result of the Stream `collect`.
```
Map<String, List<Integer>> wordsChart
```